sdk/storage/Azure.Storage.Common/swagger/Generator/src/naming.ts

```diff
@@ -46,5 +46,22 @@ export function parameter(name: string) { return variable(name); }
 export function type(name: string) { return pascalCase(name); }
 export function interfaceName(name: string) { return "I" + type(name); }
 export function namespace(name: string) { return pascalCase(name); }
-export function enumField(name: string) { return pascalCase(name); }
+export function enumField(name: string) {
+    // Convert acronyms to PascalCase
+    // example: StorageAPI -> StorageApi, StorageAPIGenerator -> StorageApiGenerator
+    if (name && name.length > 2) {
+        for (let index = 1; index < name.length; index++) {
```

Comment:
You could maybe use a regex with split to make this easier?

```
"FooBarBaz".split(/(?=[A-Z])/)
//=> ["Foo", "Bar", "Baz"]
```

Reply:
(But I don't feel strongly and you've already got working code)

Reply:
This is much simpler. Thank you :)